docs/Recursion/Count-all-subsequences-with-sum-K.md

```diff
@@ -0,0 +1,54 @@
+---
+id: count-all-subsequences-with-sum-k-problem-dsa
+title: Perfect sum
+sidebar_label: Perfect sum
+description: "The Perfect Sum problem involves finding all subsets of an array that sum up to a given target sum. The solution uses recursion and dynamic programming to efficiently count subsets with a specified sum while handling large values using modulo."
+tags: [perfect-sum, recursion, dynamic-programming, dsa]
+---
+
+## Perfect Sum Problem | Count All Subsets with Sum k
+
+- Problem Statement: Given an array arr of size n of non-negative integers and an integer sum, the task is to count all subsets of the given array with a sum equal to the target sum. Due to potentially large outputs, the answer must be computed modulo 10^9+7.
+
+- Example:
+- Input:
+  n = 6, arr = [5, 2, 3, 10, 6, 8], sum = 10
+- Output: 3
+- Explanation: The subsets that sum to 10 are {5, 2, 3}, {2, 8}, and {10}.
```

Comment:
replace 

```md
- Input:
  n = 6, arr = [5, 2, 3, 10, 6, 8], sum = 10
- Output: 3
- Explanation: The subsets that sum to 10 are {5, 2, 3}, {2, 8}, and {10}.
```

with

````md
```
- Input:
  n = 6, arr = [5, 2, 3, 10, 6, 8], sum = 10
- Output: 3
- Explanation: The subsets that sum to 10 are {5, 2, 3}, {2, 8}, and {10}.
```
````